### PR TITLE
Finite difference tests

### PR DESCRIPTION
**Pull request by cortner**
This PR transfers the FD tests for AtomsCalculators to ACU.

### CODE REVIEW

**Comment by tjjarvinen** [comment]
We can implement a calculator to wrap other calculators and conversts their output to predefined units. This would be fast to implement and solves your issue here.

**Comment by cortner** [comment]
Can I just check to be sure; your concern is that the calculator might return a different unit from uE / uL where uL is the system length unit? And that somehow leads to problems with the FD test.  (which isn't yet clear to me, but I can think it through again)

**Comment by cortner** [comment]
I believe the last two commits fix the unit conversion issue - at least in principle. Obviously there can still be bugs. Please review and merge if you are happy. 

In the meantime I will merge this into the sitepotentail branch so I continue there.

File: Project.toml [hunk 1]
```diff
@@ -7,14 +7,20 @@ version = "0.1.0-DEV"
...
+StaticArrays = "90137ffa-7385-5640-81b9-e52037218182"
 Unitful = "1986cc42-f94f-5a68-af5c-568840ba703d"
 
 [compat]
 AtomsBase = "0.3"
 AtomsCalculators = "0.1.2"
 Folds = "0.2.10"
-julia = "1.9"
 Unitful = "1"
+julia = "1.9, 1.10"
```
Comment (tjjarvinen) [inline Project.toml:20]: Having "1.9" in compat includes "1.10", so just have "1.9" and thats it.

The same thing for LinearAlgebra and Printf

**Review by tjjarvinen — COMMENTED** [review]
It is not working for me.

For this

```julia
using ORCAcalculator
using AtomsBase
using AtomsCalculators
using Unitful
using UnitfulAtomic
using AtomsCalculatorsUtilities.Testing

ox = ORCAexecutable()
om = ORCAmethod("blyp def2-TZVPP TIGHTSCF defgrid3")

h1 = isolated_system([
    :H => [0, 0, 0.]u"Å",
    :H => [0, 0, 1.]u"Å"
])

h2 = isolated_system([
    :H => auconvert.([0, 0, 0.]u"Å"),
    :H => auconvert.([0, 0, 1.]u"Å")
])

orca = ORCAcalculator.ORCAcalculatorbundle(ox, om)

##

Testing.fdtest(h1, orca; test_virial=false)  # this erros
```

I get

```julia
ERROR: AssertionError: unit((bb0[1])[1]) == uL_sys
Stacktrace:
 [1] _fdtest_forces(sys::FlexibleSystem{…}, calc::ORCAcalculatorbundle, verbose::Bool, rattle::Bool, h0::Quantity{…}; kwargs::@Kwargs{…})
   @ AtomsCalculatorsUtilities.Testing ~/Coding/Atoms/dev/AtomsCalculatorsUtilities/src/testing/fdtests.jl:106
 [2] fdtest(sys::FlexibleSystem{…}, calc::ORCAcalculatorbundle; verbose::Bool, rattle::Bool, test_virial::Bool, test_forces::Bool, tol::Float64, h0::Quantity{…}, h0_cell::Float64)
   @ AtomsCalculatorsUtilities.Testing ~/Coding/Atoms/dev/AtomsCalculatorsUtilities/src/testing/fdtests.jl:194
 [3] top-level scope
   @ ~/Coding/Atoms/local/test-fd.jl:25
Some type information was truncated. Use `show(err)` to see complete types.
```

So, I presume there is some issue(s) with units somewhere.

**Comment by cortner** [comment]
There is a compatibility test. It has to be somewhere. The system and calculator must have the same units. We can discuss where that test should be. Maybe inside the calculator

**Comment by tjjarvinen** [comment]
If it does not work for general calculator, then it does not belong to AtomsCalculatorUnitilites.

**Comment by cortner** [comment]
I violently disagree with that statement.

**Comment by cortner** [comment]
Most of the utilities are not for general calculators.

**Comment by tjjarvinen** [comment]
To calculate FD gradien you only need potential energy calculation implemented. Every calculator does that. There is no reason why it would not work for every calculator. Other than there are issues with implementation.

**Comment by cortner** [comment]
Regarding your ORCA example: the @assert that fails is actually just an assert for documentation. The units are converted to match: 
```
   uL_sys = unit(position(sys, 1)[1])
   h0 = uconvert(uL_sys, h0)
```
I suspect it's another f***ing infinity.

**Comment by tjjarvinen** [comment]
With the current version using

```julia
using ORCAcalculator
using AtomsBase
using AtomsCalculators
using Unitful
using UnitfulAtomic
using AtomsCalculatorsUtilities.Testing

ox = ORCAexecutable()
om = ORCAmethod("blyp def2-TZVPP TIGHTSCF defgrid3")

h1 = isolated_system([
    :H => [0, 0, 0.]u"Å",
    :H => [0, 0, 1.]u"Å"
])

h2 = isolated_system([
    :H => auconvert.([0, 0, 0.]u"Å"),
    :H => auconvert.([0, 0, 1.]u"Å")
])

orca = ORCAcalculator.ORCAcalculatorbundle(ox, om)

Testing.fdtest(h1, orca; test_virial=false)
```

I get

```julia
Forces finite-difference test
---------|----------- 
    h    | error 
---------|----------- 
[file orca_tools/Tool-Misc/qccc2ic.cpp, line 654]: Zero distance between atoms 2 and 1 in Cartesian2Internal

ERROR: failed process: Process(`/home/teemu/.local/opt/orca-5.0.4/orca /tmp/jl_V4xD0Q/orca_calculation.inp`, ProcessExited(18)) [18]
```

Is this related to the infinity bug?

**Comment by cortner** [comment]
I don't know what causes this. If you want me to debug it then I need a MWE. 

Btw, the FD tests need not be universal at this point. This example is not enough to prevent merging. They are useful now as they are.

**Comment by tjjarvinen** [comment]
MWE is basicly the code sniplet I posted. You need to have ORCAcalculator installed ofcourse.

**Comment by cortner** [comment]
It is not a MWE if I need to install ORCA first and then figure out how to use your ORCA calculator.

**Comment by tjjarvinen** [comment]
That is true. I will try to find out some other way to get to that error.

**Comment by cortner** [comment]
I have a MWE to reproduce this.

**Comment by cortner** [comment]
done

**Comment by tjjarvinen** [comment]
All works now!

The results are now consisten regardless of system unit type and no other errors. That documentation build issue can fixed later (need to add entry to `@autodocs` in `docs/make.jl`)

I have more than happy to merge this.

**Comment by cortner** [comment]
thank you. I will also write some docs. But I'd be grateful if you can get started on that. I.e. give them a framework and working docs build. Then I can add a bit.

**Pull request merged**